### PR TITLE
 OneToMany & ManyToMany with fully API

### PR DESCRIPTION
**Pull request by webshp62036**

### CODE REVIEW

File: Assignment1/src/main/java/com/example/assignment1/controller/EmpController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.example.assignment1.controller;
+
+import com.example.assignment1.entity.Employee;
+import com.example.assignment1.service.EmployeeService;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.ManyToOne;
+import java.util.List;
+@Controller
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/controller/EmpController.java:11]: Remove @Controller

File: Assignment1/src/main/java/com/example/assignment1/controller/EmpController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    private final EmployeeService employeeService;
+
+    public EmpController(EmployeeService employeeService) {
+        this.employeeService = employeeService;
+    }
+
+
+    // postMapping
+    @PostMapping("saveEmpData")
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/controller/EmpController.java]: API is not in camel-case

File: Assignment1/src/main/java/com/example/assignment1/controller/EmpController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public String save(@RequestBody Employee empolyee){
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/controller/EmpController.java]: create dto for data transfer

File: Assignment1/src/main/java/com/example/assignment1/entity/Employee.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    @Column(name = "emp_id")
+    private int empId;
+    @Column(name = "emp_name")
+    private String empName;
+    @Column(name = "emp_email")
+    private String empEmail;
+    @Column(name = "emp_salary")
+    private float empSalary;
+
+
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/entity/Employee.java]: remove extra spaces

File: Assignment1/src/main/java/com/example/assignment1/entity/PhoneNo.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package com.example.assignment1.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/entity/PhoneNo.java]: remove unused import

File: Assignment1/src/main/java/com/example/assignment1/entity/PhoneNo.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "p_id")
+    private int pId;
+    @Column(name = "phone_no")
+    private long phoneNo;
+    @Column(name = "zip_code")
+    private String zipCode;
+
+
+
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/entity/PhoneNo.java]: remove extra spaces

File: Assignment1/src/main/java/com/example/assignment1/repository/PhoneNoRepository.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.example.assignment1.repository;
+
+import com.example.assignment1.entity.PhoneNo;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/repository/PhoneNoRepository.java:5]: unused import

File: Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    public String saveEmpData(Employee empolyee){
+        employeeRepository.save(empolyee);
+        return "Data Successfully Save !!!!!!!!!";
+    }
+
+    public List<Employee> getEmpData() {
+        return employeeRepository.findAll();
+    }
+
+    public String updateEmpData(Employee empolyee, int empId) {
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java]: write full name updateEmloyeeData

File: Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        Employee obj =employeeRepository.findById(empId).get();
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java]: if employee doesn't exist then??

File: Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        obj.setEmpName(empolyee.getEmpName());
+        obj.setEmpEmail(empolyee.getEmpEmail());
+        obj.setEmpSalary(empolyee.getEmpSalary());
+        employeeRepository.save(obj);
+        return "Data Successfully updated !!!!!!!";
+    }
+
+    public String deleteEmpData(int empId) {
+         employeeRepository.deleteById(empId);
```
Comment (Siyarm46kashyap) [inline Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java]: first check that the employee is present in our db

Comment (webshp62036) [reply Assignment1/src/main/java/com/example/assignment1/service/EmployeeService.java]: ok i going to update all these things

**Pull request merged**